pymobiledevice3/cli/webinspector.py

```diff
@@ -295,3 +210,109 @@ def cdp(lockdown: LockdownClient, host, port):
+                                            history=FileHistory(self.webinspector_history_path()))
+
+    @abstractmethod
+    async def evaluate_expression(self, exp):
+        pass
+
+    async def js_iter(self):
+        with patch_stdout(True):
+            exp = await self.prompt_session.prompt_async(HTML('<style fg="cyan"><b>&gt;</b></style> '))
+
+        if not exp.strip():
+            return
+
+        result = await self.evaluate_expression(exp)
+        colorful_result = highlight(f'{result}', lexers.JavascriptLexer(),
+                                    formatters.TerminalTrueColorFormatter(style='stata-dark'))
+        print(colorful_result, end='')
+
+    async def js_loop(self):
+        self.start_prompt_session()
+        while True:
+            try:
+                await self.js_iter()
+            except WirError as e:
+                logger.error(e)
+            except (KeyboardInterrupt, InspectorEvaluateError):  # KeyboardInterrupt Control-C
+                pass
+            except EOFError:  # Control-D
+                return
+
+    @staticmethod
+    def webinspector_history_path() -> str:
+        return str(get_home_folder() / 'webinspector_history')
+
+
+class AutomationJsShell(JsShell):
+
+    def __init__(self, lockdown: LockdownClient, timeout: float, app: str):
+        super().__init__(lockdown, timeout, app)
+        self.automation_session = None
+        self.driver = None
+
+    async def evaluate_expression(self, exp):
+        return self.driver.execute_script(f'return {exp}')
+
+    def start_js_shell(self, url: str = ''):
+        self.automation_session = self.inspector.automation_session(self.application)
+        self.driver = WebDriver(self.automation_session)
+        try:
+            self.driver.start_session()
+            if url:
+                self.driver.get(url)
+
+            asyncio.run(super().js_loop())
+        finally:
+            self.automation_session.stop_session()
+            self.inspector.close()
+
+
+class InspectorJsShell(JsShell):
+
+    def __init__(self, lockdown: LockdownClient, timeout: float, app: str):
+        super().__init__(lockdown, timeout, app)
+        self.page = None
+        self.inspector_session = None
+
+    async def evaluate_expression(self, exp: str) -> callable:
+        return await self.inspector_session.runtime_evaluate(exp)
+
+    def query_page(self):
+        reload_pages(self.inspector)
+        available_pages = list(self.inspector.get_open_pages().get('Safari', []))
+        if not available_pages:
+            logger.error('Unable to find available pages (try to unlock device)')
+            return
+
+        page_query = [inquirer.List('page', message='choose page', choices=available_pages, carousel=True)]
+        page = inquirer.prompt(page_query, theme=GreenPassion(), raise_keyboard_interrupt=True)['page']
+        self.page = page
+
+    async def start_js_shell(self, url: str = ''):
```

Comment:
rename to `start_js_shell`

Reply:
+ keep the api similar to the AutomationJsShell

Reply:
> rename to `start`

done

Reply:
> * keep the api similar to the AutomationJsShell

what do you mean? with the `try-finally` `inspector.close`?

Reply:
done